### PR TITLE
refactor: integration test actions so they use testing.TB rather than returning errors

### PR DESCRIPTION
**Pull request by alecthomas**
This lets us leverage all of the convenient assertion functions we use in our normal tests.

### CODE REVIEW

File: integration/wrapper_test.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
+package simple_test
+
+import (
+	"fmt"
+	"testing"
+)
+
+type TestingError string
+
+// T wraps testing.TB, trapping calls to Fatalf et al and panicking. The panics are caught by
```
Comment (matt2e) [inline integration/wrapper_test.go:10]: looks like an incomplete comment

File: buildengine/build_go_test.go [hunk 2]
```diff
@@ -300,7 +300,8 @@ func TestGoModVersion(t *testing.T) {
 }
 
 func TestGeneratedTypeRegistry(t *testing.T) {
-	if !testing.Short() {
+	t.Skip("FLAKY")
```
Comment (deniseli) [inline buildengine/build_go_test.go:303]: Was this flaking in CI? I tried running it locally with increasingly absurdly high values in `go test -count=N` and couldn't reproduce a failure

Comment (deniseli) [reply buildengine/build_go_test.go:303]: We should probably make an issue to fix this and unskip the test, so we don't forget.

Comment (alecthomas) [reply buildengine/build_go_test.go:303]: Yeah it was failing very often. [This](https://github.com/TBD54566975/ftl/issues/1580) is the issue.

Comment (deniseli) [reply buildengine/build_go_test.go:303]: awesome, thanks!

**Pull request merged**